init-letsencrypt.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+read -p "Enter your domain name (example.com) : " DOMAIN_NAME
+domains=($DOMAIN_NAME evalai.$DOMAIN_NAME evalapi.$DOMAIN_NAME)
+
+touch .env
+echo "DOMAIN_NAME=$DOMAIN_NAME" > .env
+
+rsa_key_size=4096
+data_path="./certbot"
+
+# Adding a valid address is strongly recommended
+read -p "Enter your Email ID : " email
+
+staging=1 # Set to 1 if you're testing your setup to avoid hitting request limits
+
+if [ -d "$data_path" ]; then
+  read -p "Existing data found for $domains. Continue and replace existing certificate? (y/N) " decision
+  if [ "$decision" != "Y" ] && [ "$decision" != "y" ]; then
+    exit
+  fi
+fi
+
+
+if [ ! -e "$data_path/conf/options-ssl-nginx.conf" ] || [ ! -e "$data_path/conf/ssl-dhparams.pem" ]; then
+  echo "### Downloading recommended TLS parameters ..."
+  mkdir -p "$data_path/conf"
+  curl -s https://raw.githubusercontent.com/certbot/certbot/master/certbot-nginx/certbot_nginx/_internal/tls_configs/options-ssl-nginx.conf > "$data_path/conf/options-ssl-nginx.conf"
+  curl -s https://raw.githubusercontent.com/certbot/certbot/master/certbot/certbot/ssl-dhparams.pem > "$data_path/conf/ssl-dhparams.pem"
+fi
+
+echo "### Creating dummy certificate for $domains ..."
+path="/etc/letsencrypt/live/$domains"
+mkdir -p "$data_path/conf/live/$domains"
+sudo docker-compose -f docker-compose-vm.yml run --rm --entrypoint "\
+  openssl req -x509 -nodes -newkey rsa:1024 -days 1\
+    -keyout '$path/privkey.pem' \
+    -out '$path/fullchain.pem' \
+    -subj '/CN=localhost'" certbot
+
+
+echo "### Starting nginx ..."
+sudo docker-compose -f docker-compose-vm.yml up --force-recreate -d nodejs
+
+echo "### Deleting dummy certificate for $domains ..."
+sudo docker-compose -f docker-compose-vm.yml run --rm --entrypoint "\
```

Comment:
why are we creating dummy certificate and deleting it?

Reply:
you can read about that here : https://medium.com/@pentacent/nginx-and-lets-encrypt-with-docker-in-less-than-5-minutes-b4b8a60d3a71